docs/reference/builder.md

```diff
@@ -1135,6 +1135,38 @@ corresponding `ARG` instruction in the Dockerfile.
 To use these, simply pass them on the command line using the `--build-arg
 <varname>=<value>` flag.
 
+### Impact on build caching
+
+Unlike `ENV` variables, `ARG` variables are not persisted in the final image 
+that is built, however, they do impact the build cache in similar ways.
+Once an `ARG` varaible is defined, if its value is different from a previous
+build then a "cache miss" will occur upon its first usage, not its declaration.
+For example, in this Dockerfile:
+
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 RUN echo $CONT_IMG_VER
+```
+changing `CONT_IMG_VER` via the `--build-arg` command line will not cause
+a cache miss on line 2, but it will cause one at line 3. This is because
+line 2 just defines the variable and has no impact on the resulting image.
+The `RUN` on line 3 will execute the specified command and in doing so will
+define a set of environment variables (including `CONT_IMG_VER`), and its value
+might impact the resulting image, therefore a cache miss will occur.
+
+Let's look at another example:
+
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 ENV CONT_IMG_VER $CONT_IMG_VER
+4 RUN echo $CONT_IMG_VER
+```
+In this example, the cache miss will occur on line 3 because the value of
+an environment variable (which is saved in the resulting image) is changed
+as we change the `CONT_IMG_VER` build arg value.
+
```

Comment:
Should we mention that in this case, the value *is* persisted in the final image, because it is now used in an `ENV`? Possibly general knowledge, but may something that trip users.